ocs_ci/utility/kms.py

```diff
@@ -316,8 +319,9 @@ def create_ocs_vault_resources(self):
         connection_data["data"]["VAULT_ADDR"] = os.environ["VAULT_ADDR"]
         connection_data["data"]["VAULT_BACKEND_PATH"] = self.vault_backend_path
         connection_data["data"]["VAULT_CACERT"] = self.ca_cert_name
-        connection_data["data"]["VAULT_CLIENT_CERT"] = self.client_cert_name
-        connection_data["data"]["VAULT_CLIENT_KEY"] = self.client_key_name
+        if not config.ENV_DATA.get("VAULT_CA_ONLY", None):
```

Comment:
in case of VAULT_CA_ONLY, please pop the keys VAULT_CLIENT_CERT and VAULT_CLIENT_KEY from the dictionary.